### PR TITLE
Fixes #38489 - Add Remote Details page with repository table

### PR DESCRIPTION
**Pull request by pavanshekar**
#### What are the changes introduced in this pull request?

Added the Remote Details page for Flatpak Remotes, displaying detailed information along with a table of associated repositories. 

#### Considerations taken when implementing this change?

It focuses on providing a consistent user experience aligned with the existing UI patterns, ensuring accurate data display and handling various repository states. Care was taken to structure the code for maintainability and future extensibility.

#### What are the testing steps for this pull request?

1. Navigate to Content -> Flatpak remotes 
2. Click on any of the available flatpak remote 
3. You should be able to see the flatpak remote details page including the repositories table.

## Summary by Sourcery

Add a dedicated Remote Details page for Flatpak Remotes with backend support to fetch, display, and update remote metadata and list associated repositories with mirroring controls.

New Features:
- Implement a React FlatpakRemoteDetails page with breadcrumb navigation, editable URL, and a RemoteRepositoriesTable showing repository name, ID, last mirrored status, and a mirror action.
- Expose last mirrored status for Flatpak remote repositories via model methods and FlatpakRemoteMirrorStatusPresenter, and surface it in the API response.
- Add Redux actions, reducers, and selectors to fetch and update Flatpak remote details and retrieve associated repositories.
- Configure frontend routing and constants to support detail and repositories endpoints for FlatpakRemotes.

Enhancements:
- Link remote names in the Flatpak remotes index to the new details page.
- Introduce a filter_autocomplete_items hook in the autocomplete concern and exclude flatpak_remote fields from autocomplete in the remote repositories controller.
- Switch to a required organization lookup in the FlatpakRemoteRepositories API controller and extend routes to handle nested detail paths.

Tests:
- Add unit tests for FlatpakRemoteDetails component covering API calls, data rendering, and empty repository states.

### CODE REVIEW

**Comment by pavanshekar** [comment]
@sourcery-ai review

File: webpack/scenes/FlatpakRemotes/Details/FlatpakRemoteDetails.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          </TextList>
+        </TextContent>
+      </GridItem>
+
+      <GridItem span={12}>
+        <TextContent>
+          <Title headingLevel="h2" size="xl" ouiaId="flatpak-remote-subtitle">Remote repositories</Title>
+          <Text component="p" ouiaId="flatpak-remote-description" style={{ color: 'gray' }}>
+            This is the list of scanned content.
+            Mirror this content to create Satellite product repositories.
```
Comment (ianballou) [inline webpack/scenes/FlatpakRemotes/Details/FlatpakRemoteDetails.js]: `Satellite` would need to be translated, so I would reword it:

```suggestion
            This is a list of scanned flatpaks.
            Mirroring a scanned flatpak creates a repository in the product of your choice.
            Sync the repository after mirroring it from this remote to distribute its content.
```

Comment (pavanshekar) [reply webpack/scenes/FlatpakRemotes/Details/FlatpakRemoteDetails.js]: Added custom autocomplete filtering to remove flatpak_remote_id suggestions when already viewing a specific remote's repositories. Also updated the description text to clarify the flatpak remote repositories workflow as suggested.

File: app/views/katello/api/v2/flatpak_remote_repositories/base.json.rabl [hunk 2]
```diff
@@ -2,3 +2,7 @@ extends 'katello/api/v2/common/identifier'
 
 attributes :name
 attributes :label, :flatpak_remote_id
+
+node(:manifest_count) { |repo| repo.manifests.count }
+node(:tag_count) { |_repo| 1 }
```
Comment (ianballou) [inline app/views/katello/api/v2/flatpak_remote_repositories/base.json.rabl]: Why is this hard-coded to 1?

 ```suggestion
node(:tag_count) { |repo| repo.manifests.pluck(:tags).flatten.uniq.size }
```

Comment (pavanshekar) [reply app/views/katello/api/v2/flatpak_remote_repositories/base.json.rabl]: Because there will be always 1 as it only takes the latest tag

File: app/controllers/katello/api/v2/flatpak_remote_repositories_controller.rb [hunk 3]
```diff
@@ -15,6 +15,28 @@ class Api::V2::FlatpakRemoteRepositoriesController < Katello::Api::V2::ApiContro
       param :label, String, :desc => N_("Label of the flatpak remote")
     end
 
+    def auto_complete_search
```
Comment (ianballou) [inline app/controllers/katello/api/v2/flatpak_remote_repositories_controller.rb]: What do you think about refactoring FilteredAutoCompleteSearch to accept items to reject to avoid duplicating code?

Something like:

```
diff --git a/app/controllers/katello/api/v2/flatpak_remote_repositories_controller.rb b/app/controllers/katello/api/v2/flatpak_remote_repositories_controller.rb
index 8b34fb2d51..edc3a457f0 100644
--- a/app/controllers/katello/api/v2/flatpak_remote_repositories_controller.rb
+++ b/app/controllers/katello/api/v2/flatpak_remote_repositories_controller.rb
@@ -15,27 +15,7 @@ module Katello
       param :label, String, :desc => N_("Label of the flatpak remote")
     end
 
-    def auto_complete_search
-      begin
-        options = resource_class.respond_to?(:completer_scope_options) ? resource_class.completer_scope_options(params[:search]) : {}
-        items = resource_class.where(:id => self.index_relation).complete_for(params[:search], options)
-
-        items = items.reject do |item|
-          item_string = item.to_s.downcase
-          item_string.include?('flatpak_remote_id') || item_string.include?('flatpak_remote')
-        end
-
-        items = items.map do |item|
-          category = ['and', 'or', 'not', 'has'].include?(item.to_s.sub(/^.*\s+/, '')) ? _('Operators') : ''
-          part = item.to_s.sub(/^.*\b(and|or)\b/i) { |match| match.sub(/^.*\s+/, '') }
-          completed = item.to_s.chomp(part)
-          {:completed => completed, :part => part, :label => item, :category => category}
-        end
-      rescue ScopedSearch::QueryNotSupported => e
-        items = [{:error => e.to_s}]
-      end
-      render :json => items
-    end
+
 
     api :GET, "/organizations/:organization_id/flatpak_remote_repositories", N_("List flatpak remote repositories")
     api :GET, "/flatpak_remotes/:flatpak_remote_id/flatpak_remote_repositories", N_("List flatpak remote's repositories")
@@ -99,5 +79,11 @@ module Katello
       throw_resource_not_found(name: 'product', id: params[:product_id]) unless @product
       @product
     end
+
+    protected
+
+    def rejected_autocomplete_items
+      ['flatpak_remote_id', 'flatpak_remote']
+    end
   end
 end
diff --git a/app/controllers/katello/concerns/filtered_auto_complete_search.rb b/app/controllers/katello/concerns/filtered_auto_complete_search.rb
index 0fa32c1534..981a084808 100644
--- a/app/controllers/katello/concerns/filtered_auto_complete_search.rb
+++ b/app/controllers/katello/concerns/filtered_auto_complete_search.rb
@@ -9,6 +9,10 @@ module Katello
         begin
           options = resource_class.respond_to?(:completer_scope_options) ? resource_class.completer_scope_options(params[:search]) : {}
           items = resource_class.where(:id => self.index_relation).complete_for(params[:search], options)
+          
+          # Filter out rejected items
+          items = filter_autocomplete_items(items)
+          
           items = items.map do |item|
             category = ['and', 'or', 'not', 'has'].include?(item.to_s.sub(/^.*\s+/, '')) ? _('Operators') : ''
             part = item.to_s.sub(/^.*\b(and|or)\b/i) { |match| match.sub(/^.*\s+/, '') }
@@ -20,6 +24,26 @@ module Katello
         end
         render :json => items
       end
+
+      protected
+
+      # Filter out unwanted autocomplete items
+      # Override this method in controllers to provide custom filtering
+      def filter_autocomplete_items(items)
+        rejected_patterns = rejected_autocomplete_items
+        return items if rejected_patterns.empty?
+        
+        items.reject do |item|
+          item_string = item.to_s.downcase
+          rejected_patterns.any? { |pattern| item_string.include?(pattern.downcase) }
+        end
+      end
+
+      # Override this method in controllers to specify patterns to reject
+      # Returns an array of strings/patterns to filter out
+      def rejected_autocomplete_items
+        []
+      end
```

File: app/views/katello/api/v2/flatpak_remote_repositories/base.json.rabl (hunk 2 above)
Comment (ianballou) [reply app/views/katello/api/v2/flatpak_remote_repositories/base.json.rabl]: A user could theoretically remove the use of "latest" in the repo's include tags filter to sync more tags. The extra tags are available in the index to see too.
I know the client by default takes the latest tag, but that is separate from what content actually exists.
If the tag data doesn't match what is on the Index, I wonder why show it at all.

File: app/controllers/katello/api/v2/flatpak_remote_repositories_controller.rb (hunk 3 above)
Comment (pavanshekar) [reply app/controllers/katello/api/v2/flatpak_remote_repositories_controller.rb]: Agreed, this is much better! Made the suggested change and pushed it.

File: app/views/katello/api/v2/flatpak_remote_repositories/base.json.rabl (hunk 2 above)
Comment (sjha4) [reply app/views/katello/api/v2/flatpak_remote_repositories/base.json.rabl]: I think we should add a validation on mirrored flatpak repos to only allow latest tag..In the context here, we show what the mirroring does which limits tags to latest and hence the 1. I don't mind removing the field from the details here cause this is not data we store/care about outside the latest tag.

Comment (ianballou) [reply app/views/katello/api/v2/flatpak_remote_repositories/base.json.rabl]: If a user won't see the true content in the upstream index, we should remove it.

**Comment by pavanshekar** [comment]
> > I am getting an error creating bookmark for search on the remote details > repositories table.
> > ```
> > Oh no! Something went wrong while submitting the form, the server returned the following error: Controller /katello/api/v2/flatpak_remote_repositories is not a valid controller
> > ```
> 
> I wonder if this is related to how controllers are identified by the bookmarks mechanism, I remember in Foreman the controller names needed to fit a certain scheme. I had to do something special for the bootc container images page.

Yes after making this change the bookmark is working:
```diff --git a/app/lib/katello/concerns/bookmark_controller_validator_extensions.rb b/app/lib/katello/concerns/bookmark_controller_validator_extensions.rb
index e6d8d87415..545c603052 100644
--- a/app/lib/katello/concerns/bookmark_controller_validator_extensions.rb
+++ b/app/lib/katello/concerns/bookmark_controller_validator_extensions.rb
@@ -4,7 +4,7 @@ module Katello
       extend ActiveSupport::Concern
 
       def valid_controllers_list
-        @valid_controllers_list ||= (["dashboard", "common_parameters", "/katello/api/v2/host_bootc_images"] +
+        @valid_controllers_list ||= (["dashboard", "common_parameters", "/katello/api/v2/host_bootc_images", "/katello/api/v2/flatpak_remotes", "/katello/api/v2/flatpak_remote_repositories"] +
           ActiveRecord::Base.connection.tables.map(&:to_s) +
           Permission.resources.map(&:tableize)).uniq
       end```

File: webpack/scenes/FlatpakRemotes/FlatpakRemotesPage.js [hunk 4]
```diff
@@ -124,7 +126,7 @@ const FlatpakRemotesPage = () => {
                 } = remote;
                 return (
                   <Tr key={id} ouiaId={`flatpak-remote-row-${id}`}>
-                    <Td>{truncate(name)}</Td>
+                    <Td><Link to={`${urlBuilder('flatpak_remotes', '')}${id}`}>{truncate(name)}</Link></Td>
```
Comment (sjha4) [inline webpack/scenes/FlatpakRemotes/FlatpakRemotesPage.js:134]: Not sure why this doesn't register in routes. Can you use history.push to test if that works with navigation?

File: app/presenters/katello/flatpak_remote_mirror_status_presenter.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        result: @task.result,
+        started_at: @task.started_at,
+        last_mirror_words: time_ago_in_words(@task.started_at),
+      }
+    end
+
+    private
+
+    def empty_task(repo)
+      { id: repo.id, progress: {}, state: "Never synced", raw_state: "never_synced" }
```
Comment (sjha4) [inline app/presenters/katello/flatpak_remote_mirror_status_presenter.rb]:
```suggestion
      { id: repo.id, progress: {}, state: "Never mirrored or task cleaned up", raw_state: "never_mirrored" }
```

File: app/presenters/katello/flatpak_remote_mirror_status_presenter.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      return empty_task(@repo) unless @task
+
+      {
+        id: @repo.id,
+        mirror_id: @task.id,
+        state: format_state(@task),
+        raw_state: raw_state(@task),
+        result: @task.result,
+        started_at: @task.started_at,
+        last_mirror_words: time_ago_in_words(@task.started_at),
```
[the remark this answers is not in the document]
Comment (sjha4) [reply app/presenters/katello/flatpak_remote_mirror_status_presenter.rb:22]: return empty_task(@repo) unless @task handles the case. Task will always have started_at if present otherwise.

File: app/presenters/katello/flatpak_remote_mirror_status_presenter.rb [hunk 7]
```diff
@@ -0,0 +1,41 @@
+module Katello
+  class FlatpakRemoteMirrorStatusPresenter
+    include ::ActionView::Helpers::DateHelper
+    include ::Katello::TranslationHelper
+
+    def initialize(repo, task)
```
Comment (sjha4) [inline app/presenters/katello/flatpak_remote_mirror_status_presenter.rb]:
```suggestion
    def initialize(remote_repository, task)
```
Just to make the name more clear..Repo is confusing cause it could refer to repository.
